src/util.rs

```diff
@@ -209,9 +209,6 @@ impl PossibleBom {
     /// encoding. Otherwise, return `None`.
     pub fn encoding(&self) -> Option<&'static Encoding> {
         let bom = self.as_slice(true);
-        if bom.len() < 3 {
-            return None;
-        }
```

Comment:
The BOM is only 2 bytes long in UTF-16, so this always returned `None` in that case.